ggshield/cmd/iac/scan/ci.py

```diff
@@ -44,13 +48,30 @@ def scan_ci_cmd(
         return display_iac_scan_all_result(ctx, directory, result)
 
     config: Config = ctx.obj["config"]
-    commit_list, _ = collect_commit_range_from_ci_env(config.user_config.verbose)
-    reference, current_ref = commit_list[0], commit_list[-1]
 
-    # If we failed to fetch a current reference, we set it to HEAD
-    if len(commit_list) < 2 or not current_ref:
+    new_commits, _ = collect_commit_range_from_ci_env(config.user_config.verbose)
+    current_ref = new_commits[-1]
```

Comment:
Couldn't it be simpler and use a method to find the head ?
I think we should have a method that will explicitely find the `reference` and the `current` state.